Terasology/StructureTemplates
# Animate ST construction

Includes changes made by @flo and @skaldarnar in PRs #25 and #7. Also resolves various merge conflicts and alters the falling block implementation to construct buildings in a block-by-block manner. This also makes the layer-by-layer implementation the default way to construct STs.

### How to test
- Construct an existing ST, which should be done layer-by-layer in a bottom up manner.
- Add `FallingBlocksPlacementAlgorithmComponent` to an existing ST prefab. Construction of this ST will now be done by blocks falling from the sky.

Required changes in modules : 
- https://github.com/Terasology/MasterOfOreon/pull/23
- https://github.com/Terasology/DynamicCities/pull/32
- https://github.com/Terasology/CombatSystem/pull/35
- Terasology/GooeysQuests#20
- Terasology/AdventureAssets#21
- Terasology/Scenario#39
